thrift/arg2/kv_iterator.go

```diff
@@ -0,0 +1,87 @@
+package arg2
+
+import "encoding/binary"
+
+// KeyValIterator is a iterator for reading tchannel-thrift Arg2 Scheme,
+// which has key/value pairs (k~2 v~2).
+// NOTE: to be optimized for performance, we try to limit the allocation
+// done in the process of iteration.
+type KeyValIterator struct {
+	arg2Len               int
```

Comment:
do we need this and `arg2Payload`? i think we should use `len(arg2Payload)`

Reply:
Good point.